python/hail/matrixtable.py

```diff
@@ -504,14 +502,20 @@ def __getitem__(self, item):
             else:
                 col_key = wrap_to_tuple(exprs[1])
 
-            if row_key is not None and col_key is not None:
-                return self.index_entries(row_key, col_key)
-            elif row_key is not None and col_key is None:
-                return self.index_rows(*row_key)
-            elif row_key is None and col_key is not None:
-                return self.index_cols(*col_key)
-            else:
-                return self.index_globals()
+            if ((row_key is None or all(isinstance(e, Expression) for e in row_key)) and
+                (col_key is None or all(isinstance(e, Expression) for e in col_key))):
```

Comment:
can you explain this if condition to me? I don't understand what it is achieving.

Reply:
The motivation for this PR is to fix the absolutely horrendous type check errors that this method was throwing. This if statement is checking that the types are what we expect -- the row key and col key args are either `:` or a tuple of exprs. If the if fails, then we default to the usage error message.